tests/test_rabfe.py

```diff
@@ -21,6 +27,193 @@
 
 
 class TestRABFEModels(TestCase):
+    def test_endpoint_parameters_match_decoupling_and_conversion_complex(self):
+        """Verifies that the parameters at the endpoint of conversion match with the starting parameters of
+        the decoupling. Done on a complex model, as the hydration models differ
+
+        Conv: P_start -> P_independent
+        Decouple: P_independent -> P_arbitrary
+
+        """
+        host_system, host_coords, host_box, host_topology = builders.build_water_system(4.0)
+
+        num_host_atoms = host_coords.shape[0]
+
+        ff_params = hif2a_ligand_pair.ff.get_ordered_params()
+
+        temperature = 300.0
+        pressure = 1.0
+        dt = 2.5e-3
+
+        client = CUDAPoolClient(NUM_GPUS)
+
+        decouple_model = RelativeBindingModel(
+            client,
+            hif2a_ligand_pair.ff,
+            host_system,
+            construct_lambda_schedule(2),
+            host_topology,
+            temperature,
+            pressure,
+            dt,
+            10,
+            50,
+            frame_filter=all_frames,
+        )
+
+        blocker = hif2a_ligand_pair.mol_a
+        ligand = hif2a_ligand_pair.mol_b
+
+        decouple_topo = decouple_model.setup_topology(blocker, ligand)
+        decouple_ref = RelativeFreeEnergy(decouple_topo)
+
+        decouple_unbound_potentials, decouple_sys_params, _ = decouple_ref.prepare_host_edge(
+            ff_params, decouple_model.host_system
+        )
+
+        conv_model = AbsoluteConversionModel(
+            client,
+            hif2a_ligand_pair.ff,
+            host_system,
+            construct_lambda_schedule(2),
+            host_topology,
+            temperature,
+            pressure,
+            dt,
+            10,
+            50,
+            frame_filter=all_frames,
+        )
+
+        conv_topo = conv_model.setup_topology(ligand)
+        conv_ref = AbsoluteFreeEnergy(ligand, conv_topo)
+
+        conv_unbound_potentials, conv_sys_params, _ = conv_ref.prepare_host_edge(ff_params, conv_model.host_system)
```

Comment:
Observation: Dozens of lines of simulation detail (timestep, parallel client, frame filtering policy, lambda schedule, temperature, pressure, etc.) must be specified n order to obtain the `sys_params` needed for this test.

Ideally the `fe` module would allow a more direct way to prepare these -- will move this comment to https://github.com/proteneer/timemachine/issues/565

Reply:
There is likely a more direct route that could be taken already. I tested at this level as it is the level at which I think about our thermo cycle. Say the AbsoluteConversionModel changes its topology, this test would capture that while testing it at the topology level would not.

Reply:
Since the conversion/decouple only affect ligand parameters - maybe there's a way to refactor this later on to test only on the ligand?

Reply:
Though having a system level test is still a good idea!